app/frontend/components/common/asset.tsx

```diff
@@ -68,57 +85,122 @@ export const FormattedAssetItem = ({
     } else {
       return <QuestionFillIcon />
     }
-  }
+  })()
 
-  const formatAssetName = (assetNameHex: string, metadata: RegisteredTokenMetadata | undefined) => {
-    if (metadata?.name) {
-      return metadata.name
-    } else if (assetName) {
-      return assetNameHex2Readable(assetNameHex)
-    }
-    return '<no name>'
-  }
+  const {
+    labelType: formattedHumanReadableLabelType,
+    label: formattedHumanReadableLabel,
+    labelWithIcon: formattedHumanReadableLabelWithIcon,
+    shortLabelWithIcon: formattedShortHumanReadableLabelWithIcon,
+  } = (() => {
+    const {labelType, label, labelShort} = (() => {
```

Comment:
this nested function makes it quite a pain to follow what's going on. I'd consider extracting it into a `getLabelVariants` functions and the parent function, extracted (to make the `FormattedAssetItem` component easier to follow as it's getting really chunky) and renamed to `getFormattedLabelVariants` could be using it internally

Reply:
Refactored